libs/store/src/lib/infrastructure/persistence/domain/state-handler.ts

```diff
@@ -0,0 +1,47 @@
+import 'reflect-metadata';
+import { BaseValue } from '../../persistence/domain/base-classes/base-value';
+import { Type } from '../../../domain/public_api';
+
+export function instanceForType<T>(Type: Type<T>, fromState = {}) {
+    const newState = new Type(fromState);
+
+    return new Proxy(newState, new StateHandler(fromState));
+}
+/**
+ *  Proxy wrapper
+ */
+class StateHandler<TModel> {
+    constructor(
+        /** Plain entity dto object */
+        private state: TModel
+    ) {
+    }
+
+    get(target: any, prop: PropertyKey): any {
+        // Check is nested object and wraps a new one proxy
+        if (this.state[prop] && typeof this.state[prop] === 'object') {
+            return new Proxy(this.state[prop], this);
+        }
+
+        return Reflect.get(target as any, prop);
```

Comment:
Did not we say that we are not going to use Reflect? and Dima removed it from his PR as well?

Reply:
Maybe I missed something but why we shouldn't use Reflect?

Reply:
The only thing the Reflect has to use accessor option otherwise there is no much sense in it

Reply:
@fkolar it's a native Reflect, not that one that comes from `reflect-metadata`